core/unix/signal.c

```diff
@@ -1276,7 +1290,8 @@ signal_thread_exit(dcontext_t *dcontext, bool other_thread)
         }
         info->num_pending = 0;
     }
-    if (!other_thread)
+    /* If no detach flag is set, we assume that this thread is on its way to exit. */
```

Comment:
The start/stop API used to support exiting a single thread: that is no longer the case, so I think this assumption is correct, and all the temporarily-native stuff does not come here.  Is there any way we can check it via an assert or sthg?  Probably not b/c it would have to be in app code.

Reply:
If DR_EMIT_NATIVE, then I guess no way to check? What about temporarily native e.g. for rseq? In any case those paths shouldn't come here indeed? Since this is called from cleanup_and_terminate?

Reply:
(Resolving, re-open if necessary)